### PR TITLE
Fix windows prefix path for logging

### PR DESCRIPTION
**Pull request by nickgerace**
## Relevant Issue(s)

- https://github.com/rancher/rancher/issues/32268

## Note(s) to Reviewers

- For `Kube_Tag_Prefix`, capital `C` is required for this field for Windows.
  - That is why we convert `c:` to `C:` during templating.
- Due to the Kubebuilder spec, `extraVolumeMounts` must start with `/` and not `C:/`. They cannot use the `windowsInputTail` as a result. This has been tested on a Windows 20H2 cluster on k8s 1.20.6 with RKE.
  - Source: https://github.com/banzaicloud/logging-operator/blob/f18b277cbdb8b4dd4db3ac15f8dd8f50c3e28df5/pkg/sdk/api/v1beta1/node_agent_types.go#L59

## Rendered Templates

`--set global.cattle.rkeWindowsPathPrefix="C:/foo/"`
```
      nodeAgentFluentbit:
        filterKubernetes:
          Kube_Tag_Prefix: "kubernetes.C.foo.var.lib.rancher.rke.log."
        inputTail:
          Path: "C:/foo/var/lib/rancher/rke/log"
        extraVolumeMounts:
        - source: "/foo/var/lib/rancher/rke/log"
          destination: "/foo/var/lib/rancher/rke/log"
          readOnly: true
```
`--set global.cattle.rkeWindowsPathPrefix="c:\\\\foo\\\\"`
```
      nodeAgentFluentbit:
        filterKubernetes:
          Kube_Tag_Prefix: "kubernetes.C.foo.var.lib.rancher.rke.log."
        inputTail:
          Path: "C:/foo/var/lib/rancher/rke/log"
        extraVolumeMounts:
        - source: "/foo/var/lib/rancher/rke/log"
          destination: "/foo/var/lib/rancher/rke/log"
          readOnly: true
```
`--set global.cattle.rkeWindowsPathPrefix="c:/foo"`
```
      nodeAgentFluentbit:
        filterKubernetes:
          Kube_Tag_Prefix: "kubernetes.C.foo.var.lib.rancher.rke.log."
        inputTail:
          Path: "C:/foo/var/lib/rancher/rke/log"
        extraVolumeMounts:
        - source: "/foo/var/lib/rancher/rke/log"
          destination: "/foo/var/lib/rancher/rke/log"
          readOnly: true
```
`nil`
```
      nodeAgentFluentbit:
        filterKubernetes:
          Kube_Tag_Prefix: "kubernetes.C.var.lib.rancher.rke.log."
        inputTail:
          Path: "C:/var/lib/rancher/rke/log"
        extraVolumeMounts:
        - source: "/var/lib/rancher/rke/log"
          destination: "/var/lib/rancher/rke/log"
          readOnly: true
```

### CODE REVIEW

File: charts/rancher-logging/rancher-logging/3.9.401+up3.9.4/templates/_helpers.tpl [hunk 1]
```diff
@@ -75,11 +75,32 @@ true
...
+{{- end -}}
+
+{{- define "windowsInputTailMount" -}}
+{{- $temp := ((include "windowsPathPrefix" .) | replace "C:" "" | replace "c:" "") -}}
+{{- if (hasSuffix "/" $temp) -}}
+{{- trimSuffix "/" $temp -}}
+{{- else -}}
+{{- $temp -}}
+{{- end -}}
+{{- end -}}
```
Comment (nickgerace) [inline charts/rancher-logging/rancher-logging/3.9.401+up3.9.4/templates/_helpers.tpl:106]: **Note to reviewers:** Due to the Kubebuilder spec, these must start with `/`. This cannot use the `windowsInputTail` as a result. This has been tested on a Windows 20H2 cluster on k8s 1.20.6 with RKE.

Source: https://github.com/banzaicloud/logging-operator/blob/f18b277cbdb8b4dd4db3ac15f8dd8f50c3e28df5/pkg/sdk/api/v1beta1/node_agent_types.go#L59

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml [hunk 2]
```diff
@@ -27,9 +27,9 @@ spec:
       profile: windows
       nodeAgentFluentbit:
         filterKubernetes:
-          Kube_Tag_Prefix: "kuberentes.C.var.lib.rancher.rke.log."
+          Kube_Tag_Prefix: "{{ template "windowsKubernetesFilter" . }}.var.lib.rancher.rke.log."
```
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml:30]: **Note to reviewers:** capital `C` is required, which is why we convert `c:` to `C:` now.

File: packages/rancher-logging/generated-changes/patch/templates/_helpers.tpl.patch [hunk 3]
```diff
@@ -23,11 +23,32 @@
...
++{{- end -}}
++
++{{- define "windowsInputTailMount" -}}
++{{- $temp := ((include "windowsPathPrefix" .) | replace "C:" "") -}}
++{{- if (hasSuffix "/" $temp) -}}
++{{- trimSuffix "/" $temp -}}
++{{- else -}}
++{{- $temp -}}
++{{- end -}}
++{{- end -}}
```
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/patch/templates/_helpers.tpl.patch:54]: **Note to reviewers:** Due to the Kubebuilder spec, these must start with /. This cannot use the windowsInputTail as a result. This has been tested on a Windows 20H2 cluster on k8s 1.20.6 with RKE.

Source: https://github.com/banzaicloud/logging-operator/blob/f18b277cbdb8b4dd4db3ac15f8dd8f50c3e28df5/pkg/sdk/api/v1beta1/node_agent_types.go#L59

Comment (nickgerace) [inline packages/rancher-logging/generated-changes/patch/templates/_helpers.tpl.patch:54]: **Note to reviewers:** Due to the Kubebuilder spec, these must start with /. This cannot use the windowsInputTail as a result. This has been tested on a Windows 20H2 cluster on k8s 1.20.6 with RKE.

Source: https://github.com/banzaicloud/logging-operator/blob/f18b277cbdb8b4dd4db3ac15f8dd8f50c3e28df5/pkg/sdk/api/v1beta1/node_agent_types.go#L59

**Review by paynejacob — DISMISSED** [review]

**Review by paynejacob — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

**Pull request merged**